hal/src/nRF52840/timer_hal.cpp

```diff
@@ -284,6 +285,16 @@ extern "C" void RTC_IRQ_HANDLER(void) {
         sTimerMicrosAtLastOverflow = getCurrentTime();
         sTickCountAtLastOverflow = DWT->CYCCNT;
         __set_PRIMASK(pri);
+    } else if (nrf_rtc_event_pending(RTC_INSTANCE, NRF_RTC_EVENT_TICK)) {
```

Comment:
Do we know which event in particular causes this? While it is fine to just clear everything in the handler, we should have been able to do that in `hal_timer_init()` after waking up from sleep and enable only what is required.

Reply:
We make use of the `NRF_RTC_EVENT_COMPARE_0` event. Clearing events in `hal_timer_init()` doesn't work after waking up from sleep, as the interrupt handler is executed immediately following `__enable_irq();`, which is before `hal_timer_init()` being executed.

Reply:
That should not be happening though, right? We unbump the priority level of both the GPIOTE and RTC interrupts right before calling `__enable_irq()`. If they do get triggered that's a bug in the sleep code. This has also been working previously just fine for quite a while. This is not new code.

Reply:
Just to clarify: I meant that calling `__enable_irq()` at that point in sleep code does not unblock interrupts with priorities `>= 2` because we are still under effect of `BASEPRI` at that point. We do re-enable interrupts with priority `>= 5` while waiting for SoftDevice to do certain things, but that should also not apply to RTC as it's configured normally at 6.